### PR TITLE
feat: degraded state banner

### LINKED ISSUE #2883: Coinstack failure tolerance - persistent notification
### Overview

If we're dealing with a persistent failure, e.g. a coinstack is completely down, we want a way to display a persistent notification, e.g. a banner, to inform the user that the app is likely to be in a degraded state and things may not work as expected.

Requires

https://github.com/shapeshift/web/issues/2575
https://github.com/shapeshift/web/issues/2882

### References and additional details

Up to @shapeshift/product - but we can either have a single persistent notification (e.g. a banner), or small discrete notifications where things may be affected (most features - charts, asset/account pages, swapping, defi, send).

### Acceptance Criteria

It looks/feels however product decides.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

1 day

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

if we're in a persistent degraded state, we want to display a banner to the user on all screens notifying them of such, with a retry button to try refetching accounts at their convenience.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2883 

## Risk

* isolated to the app being in a degraded state, which hopefully shouldn't occur frequently

### Engineering

* check my logic is sane

### Operations

* not testable by operations

## Screenshots (if applicable)

image

image

### CODE REVIEW

File: src/state/slices/portfolioSlice/portfolioSlice.ts [hunk 1]
```diff
@@ -107,6 +107,7 @@ export const portfolioApi = createApi({
           string,
         ]
         try {
+          if (Boolean(Math.round(Math.random()))) throw new Error('fake failure')
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/portfolioSlice.ts:110]: remove before merging

File: src/context/TransactionsProvider/TransactionsProvider.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
   const assets = useSelector(selectAssets)
   const portfolioAssetIds = useSelector(selectPortfolioAssetIds)
   const accountSpecifiers = useSelector(selectAccountSpecifiers)
-  const portfolioLoadingState = useSelector(selectPortfolioLoadingStatus)
+  const portfolioLoadingStatus = useSelector(selectPortfolioLoadingStatus)
```
Comment (0xdef1cafe) [inline src/context/TransactionsProvider/TransactionsProvider.tsx:42]: missed this rename in my previous PR

**Review by gomesalexandre — DISMISSED** [review]
Tested locally, LGTM 

image


After we revert the dev testing hunk, let's get this one in 🎖️

**Review by 0xApotheosis — APPROVED** [review]
Looks good - I won't retest as only a trivial change since gomes' review.

**Pull request merged**